public/chat.js

```diff
@@ -49,27 +49,45 @@ async function connect(event) {
 
 async function sendMessage(event) {
   event.preventDefault();
-  addMessage(messageField.value)
+  addMessage(messageField.value, 'sender')
   const messageText = messageField.value;
   messageField.value = '';
   sendMessageButton.disabled = true;
+  let newMessage = ''
+
+  try {
+    const response = await prompt(messageText)
+    newMessage = response.text
+  } catch (err) {
+    if(err.type === 'insufficient_quota') {
+      newMessage = `
+        You exceeded your current openai quota. Please review your plan and billing details by visiting: 
+        <a href="https://platform.openai.com/account/usage" target="_blank">https://platform.openai.com/account/usage</a>.
+        <br/>
+        To update your apiKey, please refer to the following guidelines: 
+        <a href="https://github.com/altostra/altostra-template-gpt-web-app" target="_blank">https://github.com/altostra/altostra-template-gpt-web-app</a>.
+      `
```

Comment:
1. So this means that `newMessage` is appended unescaped. This makes me somewhat uncomfortable as I am not sure that GPT output is escaped nor valid HTML (and what if it would contain some XML?!, and what if the user asks GPT to repeat their own text, including some partial or nonsensical HTML?!)
2. Even if we keep this (unescaped messages) as a feature, an error should be treated as an error, **not** a message.... Paint everything in red (or something less scary). Block the prompt message box (so they realize they must pay attention to whatever happened). Crash their computer and put the printer on fire!!!!1!1!
3. Create a separate function for that